river/stats/test_parallel.py

```diff
@@ -9,39 +8,72 @@
 from river import stats
 
 
-@pytest.mark.parametrize(
-    "stat",
-    [
-        pytest.param(stat, id=stat.__class__.__name__)
-        for stat in [stats.Mean(), stats.Var(ddof=0), stats.Var(ddof=1)]
-    ],
-)
-def test_add(stat):
-    A = copy.deepcopy(stat)
-    B = copy.deepcopy(stat)
-    C = copy.deepcopy(stat)
+def test_add_mean():
+    A = stats.Mean()
+    B = stats.Mean()
+    C = stats.Mean()
 
     X = [random.random() for _ in range(30)]
     Y = [random.random() for _ in range(30)]
     W = [random.random() for _ in range(30)]
 
-    for x, y, w in zip(X, Y, W):
+    for i, (x, y, w) in enumerate(zip(X, Y, W)):
         A.update(x, w)
         B.update(y, w)
         C.update(x, w).update(y, w)
 
-    D = A + B
-    assert math.isclose(C.get(), D.get())
+        D = A + B
+        assert math.isclose(C.get(), D.get())
 
-    A += B
-    assert math.isclose(C.get(), A.get())
+        # TODO: clone should work instead of deepcopy
+        E = copy.deepcopy(A)
+        E += B
+        assert math.isclose(C.get(), E.get())
+
+        assert math.isclose(
+            C.get(), np.average(X[: i + 1] + Y[: i + 1], weights=W[: i + 1] + W[: i + 1])
+        )
+
+
+def _weighted_var(X, W, ddof):
+    average = np.average(X, weights=W)
+    return np.average((X - average) ** 2, weights=W) * len(X) / (len(X) - ddof)
+
+
+@pytest.mark.parametrize("ddof", [pytest.param(ddof, id=f"{ddof=}") for ddof in [0, 1, 2]])
+def test_add_var(ddof):
+    A = stats.Var(ddof)
+    B = stats.Var(ddof)
+    C = stats.Var(ddof)
+
+    X = [random.random() for _ in range(30)]
+    Y = [random.random() for _ in range(30)]
+    W = [1 for _ in range(30)]
+
+    for i, (x, y, w) in enumerate(zip(X, Y, W)):
+        A.update(x, w)
+        B.update(y, w)
+        C.update(x, w).update(y, w)
+
+        D = A + B
+        assert math.isclose(C.get(), D.get())
+
+        # TODO: clone should work instead of deepcopy
```

Comment:
`clone()` creates a fresh copy, right?

We have `clone` and `mutate`. Maybe we need to change the semantics a little bit and create a third utility:

- `clone`: creates a copy just like deepcopy
- `mutate`: mutates an existing object in place
- `??`: creates a fresh copy, just like `Class(**params)`

What do you think, @MaxHalford ?

Reply:
Good idea! I added a `include_attributes` parameter to the `clone` method. I think that's preferable to adding a third method.

Reply:
I also think this is a nice compromise :)